### ISSUE TITLE
readCurrent not working

### ISSUE THREAD

**Issue by boopanuwat** [voelkerb/STPM3X#3]
Hi, I'm use Arduino pro mini 3.3V with STPM34.
I can read voltage but current not working. why? Please help me. (I apply Current 2A and 230V )
And I try to read via UART From STPM3x Evaluation Software ,It worked.
![111](https://user-images.githubusercontent.com/79497986/220874198-c203398c-f877-4aeb-98ea-efb8acf1c9fe.JPG)

**Comment by voelkerb** [comment]
What hardware setup are you using? You can set the gain of your analog Frontend in the library [here](https://github.com/voelkerb/STPM3X/blob/55c8c50895c0c0480d0d1f6ae93bef8f8d28a64c/src/STPM3X.cpp#L1086) or use the gain correction to adjust the value if it simply has a linear relationship as out-commented in the [basic example](https://github.com/voelkerb/STPM3X/blob/55c8c50895c0c0480d0d1f6ae93bef8f8d28a64c/examples/basic/basic.ino#L26).

**Comment by raohamza888** [comment]
![image](https://github.com/voelkerb/STPM3X/assets/102152161/d0d871b9-ca1c-48da-858c-193d502b6639)
![image](https://github.com/voelkerb/STPM3X/assets/102152161/321665e0-a6e6-4600-96ed-2fe296641be7)
![image](https://github.com/voelkerb/STPM3X/assets/102152161/a658439f-64c9-4c79-8afe-11bdb12add61)

Hi, I am attempting to read data from an IC using your Arduino code. In the provided configuration, you connect the IC enable pin to digital pin 12 on your controller. However, in my case, I have not connected this pin (pin 12) to the controller; instead, I have directly connected it to VCC for operation. When I want to reset, I simply press the push button. Now, the question is, in your code, you have used pin 12 for the reset purpose. Can I ignore this, or is it essential to connect the enable pin to the controller as you did?

**Comment by voelkerb** [comment]
It is just used for a defined initial reset. CS is pulled during the reset operation. You may need to check in the datasheet if a certain reset condition is required (e.g if the chip goes to a certain mode if CS is not pulled during reset). If you have a demo board, you could simply test your approach.

**Comment by raohamza888** [comment]
Actually, I have two STM34 ICs on the hardware that I designed earlier. For both ICs' enable pins, I have created different enable circuits, taking reference from the datasheet of STPM34. I connected both of them directly to the VCC 3.3V power source at all times. If a reset is required, the user can press the push button on the hardware, and the IC will reset accordingly.

Now, in programming, I believe a reset is required for initialization or register configuration purposes. For that, I need to connect the enable circuitry to a digital pin on the microcontroller like you did in your code and define reset_pin as 12 so that I can reset it during programming or initialization. I have also shared pictures for your reference in the above message for better clarification. Now, I am planning to connect both ICs' enable pins to one microcontroller digital pin and disconnect the previously connected VCC 3.3V directly so that I can reset the ICs through the microcontroller when configuration or a reset is required.

STPM34 Datasheet: schematic, page 14/89
datasheet link: https://www.st.com/resource/en/datasheet/stpm32.pdf
![image](https://github.com/voelkerb/STPM3X/assets/102152161/1de2c4cb-6dcc-40d5-b919-47269491480c)
